### DISCUSSION TITLE
PointNet++ two questions

### DISCUSSION THREAD

**Discussion by dokato** [pyg-team/pytorch_geometric#4227] [Q&A] [answered]
I follow along the PointNet++ example:
https://github.com/pyg-team/pytorch_geometric/blob/master/examples/pointnet2_classification.py 
Looking great, but I have two questions around the specific problem I face:

1) Looks like all the pointclouds are reduced to only 1024 points `T.SamplePoints(1024)`. Can PointNett++ be applied also to data with variable number of  points?

2) After classification, is there a way to extract from the model some sort of embedding (i.e., lower-dimensional representation of my pointcloud) with a fixed size, let's say 32?

Thanks

**Comment by rusty1s** [comment, marked as answer]
1. Yes, our implementation works seamlessly with a dynamic number of points across examples.
2. Of course. You can consider returning the graph-level embedding `x` of [this line](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/pointnet2_classification.py#L59) in your `forward` function as well.

**Reply by dokato** [reply]
Great, thanks @rusty1s . BTW since we're on that, does `pytorch_geometric` implement any unsupervised methods to learn embeddings from pointclouds too?

**Reply by rusty1s** [reply]
Sadly, no. Our point cloud examples are currently supervised only, sorry :( Any pointers/links highly appreciated though :)
